modules/phase_field/src/userobjects/DiscreteNucleationInserter.C

```diff
@@ -78,13 +81,18 @@ DiscreteNucleationInserter::threadJoin(const UserObject &y)
 {
   // combine _local_nucleus_list entries from all threads on the current process
   const DiscreteNucleationInserter & uo = static_cast<const DiscreteNucleationInserter &>(y);
-  _local_nucleus_list.insert(_local_nucleus_list.end(), uo._local_nucleus_list.begin(), uo._local_nucleus_list.end());
+  _global_nucleus_list.insert(_global_nucleus_list.end(), uo._local_nucleus_list.begin(), uo._local_nucleus_list.end());
+  _changes_made += uo._changes_made;
 }
 
 void
 DiscreteNucleationInserter::finalize()
 {
-  // here we need to combine all _local_nucleus_list into a the _global_nucleus_list
-  _global_nucleus_list = _local_nucleus_list;
+  // add the _local_nucleus_list of thread zero
+  _global_nucleus_list.insert(_global_nucleus_list.end(), _local_nucleus_list.begin(), _local_nucleus_list.end());
+  // combine _global_nucleus_lists from all MPI ranks
   _communicator.allgather(_global_nucleus_list);
```

Comment:
Ok, this might be a genuine libMesh issue! I'm not getting just zeroes, but I'm getting values scrambled form other positions in my compound type that my std::vector is templated on (it is an ```std::pair<Real, Point>```).